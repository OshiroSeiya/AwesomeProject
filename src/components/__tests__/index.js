import 'react-native';
import React from 'react';
import { Provider } from 'mobx-react/native';
import renderer from 'react-test-renderer';
import Todo from '../index.js';
import TodoStore from '../../stores/todo';

describe('<Todo />', () => {
  it('renders correctly', () => {
    const stores = {
      todo: new TodoStore(),
    };
    const tree = renderer.create(
      <Provider {...stores}>
        <Todo />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('add todo renders correctly', () => {
    const stores = {
      todo: new TodoStore(),
    };

    // 一つのtodoを追加
    stores.todo.list = [
      {
        id: 'aaaaaaaa',
        message: 'test',
        isComplete: false,
      },
    ];

    const tree = renderer.create(
      <Provider {...stores}>
        <Todo />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
