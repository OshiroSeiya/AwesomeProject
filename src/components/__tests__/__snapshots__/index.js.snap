// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> add todo renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            ADD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fafafa",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            onLongPress={undefined}
            onPress={undefined}
            style={
              Array [
                Object {
                  "color": "#bfbfbf",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï‚–
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            test
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<Todo /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            ADD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
